### PR TITLE
Pinned memory

### PR DESCRIPTION
**Pull request by Rombur**
This PR is based on top of #27. It enables all the tests using pinned memory.

### CODE REVIEW

File: core/unit_test/hip/TestHIPHostPinned_Category.hpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#include <gtest/gtest.h>
+
+namespace Test {
+
+class hip_hostpinned : public ::testing::Test {
+ protected:
+  static void SetUpTestCase() {}
+
+  static void TearDownTestCase() {}
+};
```
Comment (dalg24) [inline core/unit_test/hip/TestHIPHostPinned_Category.hpp:56]: Fixture is pointless.  Remove and register test with TEST() macro

File: core/unit_test/hip/TestHIP_Spaces.cpp [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  TestViewHIPAccessible() : m_base("base", N) {}
+
+  static void run() {
+    TestViewHIPAccessible self;
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<typename MemSpace::execution_space, TagInit>(0, N),
+        self);
+    typename MemSpace::execution_space().fence();
+
+    // Next access is a different execution space, must complete prior kernel.
```
Comment (dalg24) [inline core/unit_test/hip/TestHIP_Spaces.cpp:214]: "complete prior kernel"?

File: core/unit_test/hip/TestHIP_Spaces.cpp [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    long error_count = -1;
```
Comment (dalg24) [inline core/unit_test/hip/TestHIP_Spaces.cpp:215]: Initialize to `-1`?

File: core/unit_test/hip/TestHIP_Spaces.cpp [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+namespace Test {
+
+__global__ void test_abort() { Kokkos::abort("test_abort"); }
+
+__global__ void test_hip_spaces_int_value(int *ptr) {
+  if (*ptr == 42) {
+    *ptr = 2 * 42;
+  }
+}
```
Comment (dalg24) [inline core/unit_test/hip/TestHIP_Spaces.cpp:55]: Where are these two functions used?

**Review by dalg24 — CHANGES_REQUESTED** [review]
Do not use test fixtures

File: core/unit_test/hip/TestHIP_Spaces.cpp (hunk 2 above)
Comment (Rombur) [reply core/unit_test/hip/TestHIP_Spaces.cpp:214]: The kernel just before that one initializes the value but on a different execution space (note this is true in CUDA if you have UVM, I don't think that's true with HIP unless they add UVM). So you could potentially be starting the second kernel before the previous is one is over. That's also why you have a `fence()`.

File: core/unit_test/hip/TestHIP_Spaces.cpp (hunk 3 above)
Comment (Rombur) [reply core/unit_test/hip/TestHIP_Spaces.cpp:215]: What's the problem?

Comment (dalg24) [reply core/unit_test/hip/TestHIP_Spaces.cpp:215]: Just surprised but I saw that the Cuda implementation does that as well.

**Pull request merged**